specification/provisioningservices/resource-manager/Microsoft.Devices/2017-08-21-preview/iotdps.json

```diff
@@ -0,0 +1,1663 @@
+    },
+    "name": {
+     "description": "The name of the certificate.",
+     "type": "string",
+     "readOnly": true
+    },
+    "etag": {
+     "description": "The entity tag.",
+     "type": "string",
+     "readOnly": true
+    },
+    "type": {
+     "readOnly": true,
+     "type": "string",
+     "description": "The resource type."
+    }
+   },
+   "x-ms-azure-resource": true
+  },
+  "CertificateProperties": {
+   "description": "The description of an X509 CA Certificate.",
+   "type": "object",
+   "properties": {
+    "subject": {
+     "description": "The certificate's subject name.",
+     "type": "string",
+     "readOnly": true
+    },
+    "expiry": {
+     "description": "The certificate's expiration date and time.",
+     "type": "string",
+     "format": "date-time-rfc1123",
+     "readOnly": true
+    },
+    "thumbprint": {
+     "description": "The certificate's thumbprint.",
+     "type": "string",
+     "readOnly": true
+    },
+    "isVerified": {
+     "description": "Determines whether certificate has been verified.",
+     "type": "boolean",
+     "readOnly": true
+    },
+    "created": {
+     "description": "The certificate's create date and time.",
+     "type": "string",
+     "format": "date-time-rfc1123",
+     "readOnly": true
+    },
+    "updated": {
+     "description": "The certificate's last update date and time.",
+     "type": "string",
+     "format": "date-time-rfc1123",
+     "readOnly": true
+    }
+   }
+  },
+  "IotDpsSkuInfo": {
+   "type": "object",
+   "properties": {
+    "name": {
+     "enum": [
+      "S1"
+     ],
+     "type": "string",
+     "x-ms-enum": {
+      "name": "IotDpsSku",
+      "modelAsString": "True"
+     }
+    },
+    "tier": {
+     "type": "string",
+     "readOnly": true
+    },
+    "capacity": {
+     "format": "int64",
+     "type": "integer"
+    }
+   }
+  },
+  "ProvisioningServiceDescription": {
+   "description": "The description of the Provisioning Service.",
+   "type": "object",
+   "properties": {
+    "etag": {
+     "description": "The Etag field is *not* required. If it is provided in the response body, it must also be provided as a header per the normal ETag convention.",
+     "type": "string"
+    },
+    "properties": {
+     "$ref": "#/definitions/IotDpsPropertiesDescription"
+    },
+    "sku": {
+     "$ref": "#/definitions/IotDpsSkuInfo"
+    }
+   },
+   "allOf": [
+    {
+     "$ref": "#/definitions/Resource"
+    }
+   ],
+   "required": [
+    "sku"
+   ]
+  },
+  "IotDpsPropertiesDescription": {
+   "type": "object",
+   "properties": {
+    "state": {
+     "enum": [
+      "Activating",
+      "Active",
+      "Deleting",
+      "Deleted",
+      "ActivationFailed",
+      "DeletionFailed",
+      "Transitioning",
+      "Suspending",
+      "Suspended",
+      "Resuming",
+      "FailingOver",
+      "FailoverFailed"
+     ],
+     "type": "string",
+     "x-ms-enum": {
+      "name": "Nullable`1",
```

Comment:
You can't have names like `Nullable'1`

Reply:
this has been renamed to its property name